pallets/allocations/src/lib.rs

```diff
@@ -115,19 +126,27 @@ pub mod pallet {
             let amount_for_protocol = T::ProtocolFee::get() * amount;
             let amount_for_grantee = amount.saturating_sub(amount_for_protocol);
 
-            Self::ensure_satisfy_existential_deposit(
-                &T::ProtocolFeeReceiver::account_id(),
-                amount_for_protocol,
-            )?;
-            Self::ensure_satisfy_existential_deposit(&to, amount_for_grantee)?;
-
             <CoinsConsumed<T>>::put(coins_that_will_be_consumed);
 
             T::Currency::resolve_creating(
-                &T::ProtocolFeeReceiver::account_id(),
-                T::Currency::issue(amount_for_protocol),
+                &T::PalletId::get().into_account(),
+                T::Currency::issue(amount),
             );
-            T::Currency::resolve_creating(&to, T::Currency::issue(amount_for_grantee));
+            T::Currency::transfer(
+                &T::PalletId::get().into_account(),
+                &T::ProtocolFeeReceiver::account_id(),
+                amount_for_protocol,
+                // we use `KeepAlive` here because we want the guarantee that the funds left
+                // won't be considered dust, which would prevent us from sending the rest to
+                // the grantee.
+                ExistenceRequirement::KeepAlive,
+            )?;
+            T::Currency::transfer(
+                &T::PalletId::get().into_account(),
+                &to,
+                amount_for_grantee,
+                ExistenceRequirement::AllowDeath,
```

Comment:
Perhaps we can always keep the pallet's account alive as I don't see a benefit in allowing it to die.